src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1892,6 +1892,8 @@
     <Field Name="CollectionCreation" Type="BoundExpression?" Null="allow" SkipInVisitor="true"/>
     <!-- Construct method for collection types with [CollectionBuilder]. -->
     <Field Name="CollectionBuilderMethod" Type="MethodSymbol?" Null="allow"/>
+    <!-- Conversion from construct method return type to collection type. -->
+    <Field Name="CollectionBuilderReturnTypeConversion" Type="Conversion"/>
```

Comment:
Aleksey previously removed all `Conversion` elements from the bound tree, except for `BoundConversion` itself, and then had each node hold onto an actual `BoundConversion` and a placeholder for the thing being converted, if it couldn't be directly represented. I think we should do the same here: I'd have to go dig up the changes to remember the exact reasons, but I do remember that they were related to some correctness issues.

Reply:
Not blocking the PR: Good catch. Could we add something in the BoundNodes generator to catch this in the future?

Reply:
We probably should.

Reply:
https://github.com/dotnet/roslyn/pull/69785 will validate this.